### PR TITLE
ttkit.py langauge reworked

### PR DESCRIPTION
**Pull request by comradekingu**
Not 100% sure all dashes are correct, but better anyhow.

### CODE REVIEW

File: weblate/formats/ttkit.py [hunk 1]
```diff
@@ -438,7 +438,7 @@ def is_valid_base_for_new(
         except Exception as exception:
             if errors is not None:
                 errors.append(exception)
-            report_error(cause="File parse error")
+            report_error(cause="File-parsing error")
```
Comment (orangesunny) [inline weblate/formats/ttkit.py:441]: @nijel is it possible to change this? Isn’t it gonna break something?

Comment (nijel) [reply weblate/formats/ttkit.py:441]: Yes, there is no problem with this.

File: weblate/formats/ttkit.py [hunk 2]
```diff
@@ -645,8 +645,8 @@ def is_translated(self):
     def is_fuzzy(self, fallback=False):
         """Check whether unit needs edit.
 
-        The isfuzzy on XLIFF is really messing up approved flag with fuzzy
-        and leading to various problems.
+        The isfuzzy on XLIFF is really messing up \"approved"\\ flag with fuzzy
```
Comment (nijel) [inline weblate/formats/ttkit.py:648]: There is no need to escape quotes here:

```suggestion
        The isfuzzy on XLIFF is really messing up "approved" flag with fuzzy
```

File: weblate/formats/ttkit.py [hunk 3]
```diff
@@ -329,7 +329,7 @@ def is_valid(self):
         """Check whether store seems to be valid.
 
         In some cases Translate Toolkit happily "parses" the file, even though it really
-        did not do so (e.g. gettext parser on random text file).
+        did not do so (e.g. gettext parser on random textfile).
```
Comment (Fat-Zer) [inline weblate/formats/ttkit.py:332]:
```suggestion
        did not do so (e.g. gettext parser on a random textfile).
```

File: weblate/formats/ttkit.py [hunk 4]
```diff
@@ -189,7 +189,7 @@ def target(self):
     def context(self):
         """Return context of message.
 
-        In some cases we have to use ID here to make all backends consistent.
+        In some cases we have to use ID here to make all back-ends consistent.
```
Comment (Fat-Zer) [inline weblate/formats/ttkit.py:192]:
```suggestion
        In some cases we have to use ID here to make all the back-ends consistent.
```

File: weblate/formats/ttkit.py [hunk 5]
```diff
@@ -89,7 +89,7 @@ def target(self):
...
```
Comment (Fat-Zer) [inline weblate/formats/ttkit.py:92]:
```suggestion
        In some cases we have to use ID here to make all the back-ends consistent.
```

File: weblate/formats/ttkit.py [hunk 6]
```diff
@@ -68,7 +68,7 @@
 class TTKitUnit(TranslationUnit):
     @cached_property
     def locations(self):
-        """Return comma separated list of locations."""
+        """Return comma-separated list of locations."""
```
Comment (Fat-Zer) [inline weblate/formats/ttkit.py:71]:
```suggestion
        """Return a comma-separated list of locations."""
```

File: weblate/formats/ttkit.py [hunk 7]
```diff
@@ -16,7 +16,7 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
 #
-"""Translate Toolkit based file format wrappers."""
+"""Translate Toolkit-based file-format wrappers."""
```
Comment (Fat-Zer) [inline weblate/formats/ttkit.py:19]: it's "based upon Translate Toolkit", not "to Translate... Toolkit-based". Better leave it without a dash or use something like this:

```suggestion
"""TranslateToolkit-based file-format wrappers."""
```

File: weblate/formats/ttkit.py [hunk 8]
```diff
@@ -451,11 +451,11 @@ def delete_unit(self, ttkit_unit) -> Optional[str]:
 
 
 class PropertiesUnit(KeyValueUnit):
-    """Wrapper for properties based units."""
+    """Wrapper for properties-based units."""
 
     @cached_property
     def locations(self):
-        """Return comma separated list of locations."""
+        """Return comma-separated list of locations."""
```
Comment (Fat-Zer) [inline weblate/formats/ttkit.py:458]:
```suggestion
        """Return a comma-separated list of locations."""
```

File: weblate/formats/ttkit.py [hunk 9]
```diff
@@ -519,7 +519,7 @@ class PoMonoUnit(PoUnit):
     def context(self):
         """Return context of message.
 
-        In some cases we have to use ID here to make all backends consistent.
+        In some cases we have to use ID here to make all back-ends consistent.
```
Comment (Fat-Zer) [inline weblate/formats/ttkit.py:522]:
```suggestion
        In some cases we have to use ID here to make all the back-ends consistent.
```

File: weblate/formats/ttkit.py [hunk 10]
```diff
@@ -645,8 +645,8 @@ def is_translated(self):
     def is_fuzzy(self, fallback=False):
         """Check whether unit needs edit.
 
-        The isfuzzy on XLIFF is really messing up approved flag with fuzzy
-        and leading to various problems.
+        The isfuzzy on XLIFF is really messing up "approved" flag with fuzzy
+        strings, leading to various problems.
```
Comment (Fat-Zer) [inline weblate/formats/ttkit.py:649]: This one is incorrect, the string supposed to mean: .. is messing up the "approved" flag with the "fuzzy" flag

File: weblate/formats/ttkit.py [hunk 11]
```diff
@@ -720,7 +720,7 @@ def source(self):
 
     @cached_property
     def locations(self):
-        """Return comma separated list of locations."""
+        """Return comma-separated list of locations."""
```
Comment (Fat-Zer) [inline weblate/formats/ttkit.py:723]:
```suggestion
        """Return a comma-separated list of locations."""
```

**Review by Fat-Zer — CHANGES_REQUESTED** [review]

**Comment by Fat-Zer** [comment]
Note: I'm not really sure about the articles suggestions here... so feel free to ignore those.

File: weblate/formats/ttkit.py (hunk 7 above)
Comment (orangesunny) [reply weblate/formats/ttkit.py:19]: keeping original

File: weblate/formats/ttkit.py (hunk 9 above)
Comment (orangesunny) [reply weblate/formats/ttkit.py:522]: solved

File: weblate/formats/ttkit.py (hunk 10 above)
Comment (orangesunny) [reply weblate/formats/ttkit.py:649]: yes, corrected

**Comment by orangesunny** [comment]
> Note: I'm not really sure about the articles suggestions here... so feel free to ignore those.

some of them were useful! Thanks.

**Pull request merged**